cdap-master/src/main/java/co/cask/cdap/data/runtime/main/MasterServiceMain.java

```diff
@@ -481,96 +487,28 @@ private void updateConfigurationTable() {
-      htd.addFamily(new HColumnDescriptor(Constants.ReplicationStatusTool.WRITE_TIME));
-      htd.addFamily(new HColumnDescriptor(Constants.ReplicationStatusTool.REPLICATE_TIME));
-      htd.addFamily(new HColumnDescriptor(Constants.ReplicationStatusTool.SHUTDOWN_TIME));
-      tableUtil.createTableIfNotExists(admin, tableId, htd.build());
-      // Clear shutdown row to keep shutdown time for current instance only
-      saveShutdownTime(0L);
-    } catch (IOException e) {
-      throw Throwables.propagate(e);
-    } finally {
-      admin.close();
-    }
-    LOG.info("replicationstate table created");
-  }
-
-  private TableId getReplicationStateTableId(HBaseTableUtil tableUtil) throws IOException {
-    String tableName = hConf.get(Constants.ReplicationStatusTool.REPLICATION_STATE_TABLE_NAME);
-    String ns = hConf.get(Constants.ReplicationStatusTool.REPLICATION_STATE_TABLE_NAMESPACE);
-
-    TableId tableId = tableUtil.createHTableId(
-      (ns != null)
-        ? new NamespaceId(ns)
-        : new NamespaceId(Constants.ReplicationStatusTool.REPLICATION_STATE_TABLE_DEFAULT_NAMESPACE),
-      (tableName != null) ? tableName : Constants.ReplicationStatusTool.REPLICATION_STATE_TABLE_DEFAULT_NAME);
-
-    return tableId;
+  private static boolean isReplication(final Configuration c) {
+    return c.getBoolean(REPLICATION_ENABLE_KEY, HConstants.REPLICATION_ENABLE_DEFAULT);
   }
 
   /**
    * The replication Status tool will use CDAP shutdown time to determine last CDAP related writes to HBase.
    */
-  private void saveShutdownTime(long timestamp) {
-    HBaseAdmin admin = null;
-    HTable table = null;
+  private void saveShutdownTime(Long timestamp) {
+    if (!isReplication(hConf)) {
+      return;
+    }
     try {
-      admin = new HBaseAdmin(hConf);
-      HBaseTableUtil tableUtil = new HBaseTableUtilFactory(cConf).get();
-      TableId tableId = getReplicationStateTableId(tableUtil);
-      table = tableUtil.createHTable(hConf, tableId);
-
-      Put put = getShutdownRowPut(timestamp);
-      table.put(put);
-
-      LOG.info("Saved CDAP shutdown time = " + timestamp + " to replicationstate Table");
-    } catch (IOException ioe) {
-      LOG.error("Add CDAP shutdown time to Replication-State table Failed.");
-      Throwables.propagate(ioe);
-    } finally {
-      if (admin != null) {
-        try {
-          admin.close();
-        } catch (IOException ioe) {
-          LOG.error("Error closing HBaseAdmin: ", ioe);
-        }
-      }
-      if (table != null) {
-        try {
-          table.close();
-        } catch (IOException ioe) {
-          LOG.error("Error closing Table: " + ioe.getMessage(), ioe);
-        }
-      }
+      FileWriter fw = new FileWriter(Constants.Replication.CDAP_SHUTDOWN_TIME_FILENAME);
+      BufferedWriter bw = new BufferedWriter(fw);
+      //Write shutdown time
+      bw.write(timestamp.toString());
+      bw.close();
```

Comment:
close should usually be called in a finally block. You can also do:

```
try (BufferedWriter bw = new BufferedWriter(new FileWriter(...))) {
}
```
which will do it automatically.